### PR TITLE
fix: validatorupdateblock update every block cause messy of upgrade

### PR DESCRIPTION
**Pull request by leonz789**
## Description

When a node restart and the validatorset has just been update at the height when this node stopped, the recahe process might be failed to fill params into aggregatorContext.

## Changes
Notable changes
- Set params for aggregatorContext in recache process when validatorset changed before current height

----

Closes #XXX

### CODE REVIEW

File: x/oracle/module.go [hunk 1]
```diff
@@ -216,6 +216,7 @@ func (am AppModule) EndBlock(ctx sdk.Context, _ abci.RequestEndBlock) []abci.Val
 	logger.Info("prepare for next oracle round of each tokenFeeder")
 	agc.PrepareRound(ctx, 0)
 
+	// TODO: update params happened during this block for cache, for the case: agc is recached from history and cache'params is set with the latest params by recache, but parmas changed during this block as well. or force agc to be GET first before cache be GET(later approch is better)
```
Comment (MaxMustermann2) [inline x/oracle/module.go:219]: Will this be resolved within this PR?

**Comment by cloud8little** [comment]
Test passed. 

test object: https://github.com/ExocoreNetwork/exocore/pull/57/commits/79d675d35de95f7128146a18fce15f81b1e54dfb

Steps: 
1. For the broken nodes, use the binary with this fix, for the non-broken nodes, use the PR #56 binary. 
2. Start all nodes, all nodes start consensus.
3. after all nodes reach to the same height. stop all nodes.
4. upgrade all nodes to use the binary with #57, start all nodes, all nodes works well.

**Review by cloud8little — APPROVED** [review]

File: x/oracle/module.go (hunk 1 above)
Comment (leonz789) [reply x/oracle/module.go:219]: This is going to be implemented in a separate PR

**Review by mikebraver — APPROVED** [review]
LGTM

**Review by MaxMustermann2 — APPROVED** [review]

**Pull request merged**